config/config.go

```diff
@@ -628,7 +629,19 @@ func (c *RemoteWriteConfig) UnmarshalYAML(unmarshal func(interface{}) error) err
 	// The UnmarshalYAML method of HTTPClientConfig is not being called because it's not a pointer.
 	// We cannot make it a pointer as the parser panics for inlined pointer structs.
 	// Thus we just do its validation here.
-	return c.HTTPClientConfig.Validate()
+	if err := c.HTTPClientConfig.Validate(); err != nil {
+		return err
+	}
+
+	httpClientConfigEnabled := c.HTTPClientConfig.BasicAuth != nil ||
+		len(c.HTTPClientConfig.BearerToken) > 0 ||
+		len(c.HTTPClientConfig.BearerTokenFile) > 0
```

Comment:
```suggestion
	httpClientConfigEnabled := c.HTTPClientConfig.BasicAuth != nil ||
		c.HTTPClientConfig.Authorization != nil 
```

This is now changed in master.